### PR TITLE
add IntelliJ developer setup HowTo

### PR DESCRIPTION
**Pull request by pravussum**
Adds an initial HowTo to setup IntelliJ IDE for addon development
Incorporated feedback from https://community.openhab.org/t/howto-setup-intellij-for-binding-development/87135.

### CODE REVIEW

File: developers/ide/intellij.md [hunk 1]
```diff
@@ -5,7 +5,37 @@ title: IntelliJ
 
 # IntelliJ IDE
 
-::: tip TODO
-We are currently reworking how to setup a perfect development enviroment.
-A new step by step guide will appear here soon.
-:::
+### Prerequisities
```
Comment (Confectrician) [inline developers/ide/intellij.md]: Please use correct headings hierarchy.

```suggestion
## Prerequisities
```

File: developers/ide/intellij.md [hunk 2]
```diff
@@ -5,7 +5,37 @@ title: IntelliJ
...
+  - git, Maven, IntelliJ and Java 8 are installed
+
+### Install OpenHAB distribution
```
Comment (Confectrician) [inline developers/ide/intellij.md]: Please use correct headings hierarchy.

File: developers/ide/intellij.md [hunk 3]
```diff
@@ -5,7 +5,37 @@ title: IntelliJ
...
+1. Install the official [openHAB distribution](https://www.openhab.org/download/)
+1. Start the distribution **in debug mode** (use `./start_debug.sh` instead of `./start.sh` in step 4)
+
+I will refer to the directory where you installed the distribution as <DISTRO_DIR>.
+
+### Build the addons repostory
```
Comment (Confectrician) [inline developers/ide/intellij.md]: Please use correct headings hierarchy.

File: developers/ide/intellij.md [hunk 4]
```diff
@@ -5,7 +5,37 @@ title: IntelliJ
...
+1. fork and clone the [openhab addons repository](http://www.github.com/openhab/openhab2-addons) into a new directory (I'll refer to by <ADDON_DIR> from now on)
+    - `git clone https://github.com/<yourgitusername>/openhab2-addons`  (replace git user name accordingly)
+1. open IntelliJ and create a new Project from existing sources (File | New | Project from existing sources) and pick <ADDON_DIR>/pom.xml
+    - IntelliJ will start importing, indexing and building, it will take while, wait until finished (see status bar)
+1. Use Maven to clean & install the addons project
+    - mvn clean install in the root of <ADDON_DIR> using commandline Maven (or IntelliJ Maven view)
+    - some of the addons might fail to build - if it's not the one, you're interested in that should not bother you
+    - when the Maven project finished, you should find the freshly built addon JAR in the target directory
+
+### Debug your addon
```
Comment (Confectrician) [inline developers/ide/intellij.md]: Please use correct headings hierarchy.

File: developers/ide/intellij.md [hunk 5]
```diff
@@ -5,7 +5,37 @@ title: IntelliJ
...
-A new step by step guide will appear here soon.
-:::
+### Prerequisities
+  - git, Maven, IntelliJ and Java 8 are installed
+
+### Install OpenHAB distribution
+1. Install the official [openHAB distribution](https://www.openhab.org/download/)
+1. Start the distribution **in debug mode** (use `./start_debug.sh` instead of `./start.sh` in step 4)
+
+I will refer to the directory where you installed the distribution as <DISTRO_DIR>.
```
Comment (Confectrician) [inline developers/ide/intellij.md]:
```suggestion
This article refers to the directory where you installed the distribution as `<DISTRO_DIR>`.
```

File: developers/ide/intellij.md [hunk 6]
```diff
@@ -5,7 +5,37 @@ title: IntelliJ
...
+
+### Build the addons repostory
+1. fork and clone the [openhab addons repository](http://www.github.com/openhab/openhab2-addons) into a new directory (I'll refer to by <ADDON_DIR> from now on)
```
Comment (Confectrician) [inline developers/ide/intellij.md]:
```suggestion
1. fork and clone the [openhab addons repository](http://www.github.com/openhab/openhab-addons) into a new directory (Reference `<ADDON_DIR>` from now on for this arcticle)
```

File: developers/ide/intellij.md [hunk 7]
```diff
@@ -5,7 +5,37 @@ title: IntelliJ
...
+    - `git clone https://github.com/<yourgitusername>/openhab2-addons`  (replace git user name accordingly)
```
Comment (Confectrician) [inline developers/ide/intellij.md]: Repository name changed.

```suggestion
    - `git clone https://github.com/<yourgitusername>/openhab-addons`  (replace git user name accordingly)
```

File: developers/ide/intellij.md [hunk 8]
```diff
@@ -5,7 +5,37 @@ title: IntelliJ
...
+1. open IntelliJ and create a new Project from existing sources (File | New | Project from existing sources) and pick <ADDON_DIR>/pom.xml
```
Comment (Confectrician) [inline developers/ide/intellij.md]:
```suggestion
1. open IntelliJ and create a new Project from existing sources (File | New | Project from existing sources) and pick `<ADDON_DIR>`/pom.xml
```

File: developers/ide/intellij.md [hunk 9]
```diff
@@ -5,7 +5,37 @@ title: IntelliJ
...
+    - IntelliJ will start importing, indexing and building, it will take while, wait until finished (see status bar)
+1. Use Maven to clean & install the addons project
+    - mvn clean install in the root of <ADDON_DIR> using commandline Maven (or IntelliJ Maven view)
```
Comment (Confectrician) [inline developers/ide/intellij.md]:
```suggestion
    - mvn clean install in the root of `<ADDON_DIR>` using commandline Maven (or IntelliJ Maven view)
```

File: developers/ide/intellij.md [hunk 10]
```diff
@@ -5,7 +5,37 @@ title: IntelliJ
...
+1. create a Remote Debug runtime configuration in IntelliJ:
+    - Open menu Run | Edit configurations
+    - click the + sign to add a "Remote" configuration
+    - adapt the module setting to the root (org.openhab.addons.reactor)
+    - click OK
+    - start the debug configuration
+    - the IntelliJ console should log: `Connected to the target VM, address: 'localhost:5005', transport: 'socket'`
+![Remote Debug Run Configuration](images/ide_setup_intellij_debug_configuration.png)
+
+You can now add breakpoints to your project now and your test distro should stop there...
```
Comment (Confectrician) [inline developers/ide/intellij.md]:
```suggestion
You can now add breakpoints to your project now and your test distro should stop there.

```

**Comment by Confectrician** [comment]
Please also note my faq about dco check here: https://community.openhab.org/t/dco-check-signing-off-with-github-web-editor-explanation/83330?u=confectrician

It is enough to add the sign-of fto each commit.
I will squash commits on merge anyway, so usually you would not have to force push every time.
(As long as the sign off statement is included in every commit.)

Also the sign of with your real email adress should be the correct one.

**Comment by pravussum** [comment]
OK, I incorporated the changes. Had to fiddle a bit to get the sign off working - it also checks, wether the author/email of the commiter matches the sign-off data. Your FAQ helped, though. 
Quite some hassle for a simple documentation change :stuck_out_tongue: 
Thanks for the quick review!

**Comment by Confectrician** [comment]
>Quite some hassle for a simple documentation change 😛

I know that feeling. 😄 
The DCO check is activated organisation-wide, which includes the docs besides the code related repos where it makes more sense for sure.
That's the reason, why we handle it very loose here with the small patch exception.
Nearly half of the pull requests here have overridden dco checks from me.

On the other hand we have a big amount of contributors who are contributing code also,
so they are simply used to signing off already.

**Pull request merged**